app/assets/javascripts/app/idv-finance-helper.js

```diff
@@ -2,13 +2,50 @@ document.addEventListener('DOMContentLoaded', () => {
   const financeCntnr = document.querySelector('.js-finance-choice-cntnr');
 
   if (financeCntnr) {
+    const inputWrappers = document.querySelectorAll('.js-finance-wrapper');
+    hideAll(inputWrappers);
+
+    const currentRadio = financeCntnr.querySelector('input[type="radio"][checked]');
     const radios = financeCntnr.querySelectorAll('input[type="radio"]');
-    const label = document.querySelector('.js-finance-label');
+
+    showInput(currentRadio || radios[0]);
 
     Array.prototype.forEach.call(radios, (el) => {
       el.addEventListener('change', () => {
-        label.textContent = el.parentElement.textContent;
+        if (el.checked) {
+          showInput(el);
+        }
       });
     });
   }
+
+  function showInput(radio) {
+    const inputWrappers = document.querySelectorAll('.js-finance-wrapper');
+    hideAll(inputWrappers);
+
+    const financeType = radio.value;
+    const inputWrapperToShow = document.querySelector(`[data-type="${financeType}"]`);
+    removeClass(inputWrapperToShow, 'hide');
+  }
+
+  function hideAll(elems) {
+    Array.prototype.forEach.call(elems, (el) => {
+      addClass(el, 'hide');
+    });
+  }
+
+  function addClass(elem, klass) {
```

Comment:
thoughts on including this tiny, robust library which gives u the add/remove class functionality instead of these `addClass` and `removeClass` methods?

https://www.npmjs.com/package/classlist.js